app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/UserPointTest.java

```diff
@@ -0,0 +1,87 @@
+package com.github.ravifrancesco.softwaredevproject;
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.rule.ServiceTestRule;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Random;
+import java.util.concurrent.TimeoutException;
+
+
+public class UserPointTest {
+
+    @Rule
+    public final ServiceTestRule serviceRule = new ServiceTestRule();
+
+    @Test
+    public void getLocationTest() throws TimeoutException {
```

Comment:
What is the difference between getLocationTest and testGPSTracker? Both tests depend on the userPoint right?
Also add some comments for the tests

Reply:
It just checks that the gpstracker passes the location correctly to the userPoint